### PR TITLE
Java11 upgrade

### PR DESCRIPTION
**Pull request by nyssen**
Upgrade to OpenJDK 11 and OpenJFX 11 and Xtext/Xtend 2.17.

### CODE REVIEW

**Comment by miklossy** [comment]
I took a look into these changes and I think it is ready to be merged into the master branch. Some small adjustments (e.g. addressing deprecation warnings, get rid of the non-used generated stubs, get rid of the non-used `*Parsing` test classes) can be later performed after the merge.

File: org.eclipse.gef.dot/src/org/eclipse/gef/dot/internal/language/parser/antlr/lexer/Lexer.java [hunk 1]
```diff
@@ -26,8 +26,7 @@
 /**
  * This derived class includes the fix
  * https://github.com/eclipse/xtext-core/commit/34d61d4d80c6992045ed013a0ac4eb337a1f4e87
- * available from Xtext Version 2.9 (MARS). TODO: drop this class as soon as
- * MARS will be the minimum supported platform.
+ * available from Xtext Version 2.9 (MARS).
```
Comment (wntrzo) [inline org.eclipse.gef.dot/src/org/eclipse/gef/dot/internal/language/parser/antlr/lexer/Lexer.java:29]: Is there a reason for not removing this class?

File: org.eclipse.gef.dot.tests/src/org/eclipse/gef/dot/tests/test.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+dotpath=/usr/local/bin/dot
```
Comment (wntrzo) [inline org.eclipse.gef.dot.tests/src/org/eclipse/gef/dot/tests/test.properties:1]: Is this meant to be hardcoded?

**Comment by wntrzo** [comment]
I looked into this as well.

I couldn't do a manual test of DotGraphView yet as I couldn't manage the runtime Eclipse to find the javafx libs. You mentioned a setting that needed to be passed in run configurations, @nyssen ?

Otherwise I raised the following questions:
-the target definitions for 2019-09 and 2019-12 still need to be changed
-there is a dotpath in test.properties. Is it intended to hard-code this?
-as @miklossy mentioned, we should remove the autogenerated Parsing tests or move tests into them. Would we be able to change the Name of the Autogenerated Files to accept our Test files e.g. DotPortPosTest instead of DotPortPosParsingTest? Alternatively, we could rename our old Test classes accordingly.
-for clarification: why have we increased the dot version to from 5.0.500 to 5.1.0 instead of 5.3.0 to 5.4.0, or is the dot version unrelated to the gef version?
-There are a few todos to remove workarounds in the dot code I believe. Do we need to remove them now or should we do this at a later stage? See also my comment regarding  org.eclipse.gef.dot/src/org/eclipse/gef/dot/internal/language/parser/antlr/lexer/Lexer.java.
-The 30 minute Travis timeout could probably be reversed with @miklossy 's commit
-we need to fix copyright

Should I help removing the conflicts due to my earlier commits in master?
->the master pom in this branch (travis works for whichever reason) still uses the nonexistent "no_deps" package. (I double checked the mirror, it's still not there)...


Otherwise, I agree with Tamás, it seems to be working as far as I could test it.

Only thing: adding the new IDE project broke the build settings in my original workspace, so I had to set up a new one after checking out the branch. I suppose this issue sat a couple of inches in front of the screen though ;).

**Comment by miklossy** [comment]
Yes, it would be awesome if you could help us resolving the conflicts, after that merge this into master and then do the clean up work.

**Comment by mwienand** [comment]
> Yes, it would be awesome if you could help us resolving the conflicts, after that merge this into master and then do the clean up work.

Hm... we should NOT merge into master just yet.

After resolving, and cleaning up, we need to prepare Jenkins to build with Java 11. Webmaster did not yet install OpenJDK 11 and OpenJFX 11 SDK.

**Comment by mwienand** [comment]
Tomorrow, Tamas and I are working on GEF. We will have a closer look into this PR, too.

**Comment by wntrzo** [comment]
I had a look into Dot Graph view with all my example files. Seems to be running fine.

File: org.eclipse.gef.dot/src/org/eclipse/gef/dot/internal/language/parser/antlr/lexer/Lexer.java (hunk 1 above)
Comment (miklossy) [reply org.eclipse.gef.dot/src/org/eclipse/gef/dot/internal/language/parser/antlr/lexer/Lexer.java:29]: This class can be removed after the merge. However, first we should wait for that the GEF Jenkins Jobs are running again (`Bugzilla 559286`) to see which Xtext version will be the minimum required by GEF DOT.

File: org.eclipse.gef.dot.tests/src/org/eclipse/gef/dot/tests/test.properties (hunk 2 above)
Comment (miklossy) [reply org.eclipse.gef.dot.tests/src/org/eclipse/gef/dot/tests/test.properties:1]: We have already removed the `test.properties` file after the merge. That had been checked in by accident.

**Review by miklossy — COMMENTED** [review]
> -as @miklossy mentioned, we should remove the autogenerated Parsing tests or move tests into them. Would we be able to change the Name of the Autogenerated Files to accept our Test files e.g. DotPortPosTest instead of DotPortPosParsingTest? Alternatively, we could rename our old Test classes accordingly.

I already modified the `GenerateDot.mwe2` workflow not to generate the *ParsingTest.xtend files and also removed them from the repository.

**Comment by miklossy** [comment]
> -There are a few todos to remove workarounds in the dot code I believe. Do we need to remove them now or should we do this at a later stage? See also my comment regarding org.eclipse.gef.dot/src/org/eclipse/gef/dot/internal/language/parser/antlr/lexer/Lexer.java.

First we should wait for that the GEF Jenkins Jobs are running again (Bugzilla 559286) to see which Xtext version will be the minimum required by GEF DOT. After that, we can refactor the GEF DOT code base to remove the workarounds that are not needed anymore and also address the deprecations.

**Pull request merged**